evennia/contrib/worldloader/command.py

```diff
@@ -0,0 +1,281 @@
+                caller.msg("%s imported." % model_name)
+                count += 1
+            except Exception, e:
+                print e
+                continue
+
+        caller.msg("total %d files imported." % count)
+
+
+#------------------------------------------------------------
+# set object's info_db and info_key
+#------------------------------------------------------------
+class CmdSetDataInfo(MuxCommand):
+    """
+    Usage:
+    @datainfo <obj>[=<key>]
+    
+    This will set the data key to an object.
+    @datainfo <obj> will show the data key of the object.
+    """
+    key = "@datainfo"
+    locks = "perm(Builders)"
+    help_cateogory = "Building"
+    
+    def func(self):
+        """
+        Implement the command
+        """
+        caller = self.caller
+        if not self.args:
+            string = "Usage: @datainfo <obj>[=<key>]"
+            caller.msg(string)
+            return
+        
+        if not self.rhs:
+            if self.args == self.lhs:
+                # no "="
+                obj_name = self.args
+                obj = caller.search(obj_name, location=caller.location)
+                if not obj:
+                    caller.msg("Sorry, can not find %s." % obj_name)
+                elif obj.db.info_db and obj.db.info_key:
+                    caller.msg("%s's datainfo is %s" % (obj_name, obj.db.info_db + "." + obj.db.info_key))
+                else:
+                    caller.msg("%s has no datainfo." % obj_name)
+                return
+
+        obj_name = self.lhs
+        obj = caller.search(obj_name, location=caller.location)
+        if not obj:
+            caller.msg("Sorry, can not find %s." % obj_name)
+            return
+
+        # set the key:
+        key_name = self.rhs
+        model_name = ""
+        app_name = ""
+        
+        if key_name:
+            for model in world_settings.WORLD_DATA:
+                model_obj = get_model(world_settings.WORLD_DATA_APP, model)
+                if model_obj:
+                    if model_obj.objects.filter(key=key_name):
+                        model_name = model
+                        app_name = world_settings.WORLD_DATA_APP
+                        break
+
+        try:
+            set_obj_data_info(obj, app_name, model_name, key_name)
+            caller.msg("%s's datainfo has been set to %s" % (obj_name, self.rhs))
+        except Exception, e:
+            caller.msg("Can't set datainfo %s to %s: %s" % (self.rhs, obj_name, e))
+
+
+#------------------------------------------------------------
+# batch builder
+#------------------------------------------------------------
+class CmdBatchBuilder(MuxCommand):
+    """
+    Usage:
+      @batchbuilder
+      
+    Build the whole game world with data in CSV files.
+    """
+    key = "@batchbuilder"
+    aliases = ["@batchbld"]
+    locks = "perm(Builders)"
+    help_cateogory = "Builders"
+    arg_regex = r"\s.*?|$"
+
+    def func(self):
```

Comment:
This seems like a very limited command - you can only ever use it once until someone changes the settings. Would it not be more reasonable for this command to take a CSV file as input and build using it?